tests/spawner/test_random_schedule_strategy.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from src.schedule.random_schedule_strategy import RandomScheduleStrategy
+from tests.decorators import recreate_db_setup
+
+
+class TestRandomScheduleStrategy:
+    """Test the random schedule strategy"""
+
+    @recreate_db_setup
+    def setup_method(self):
+        pass
+
+    @pytest.mark.usefixtures(
+        "random_strategy", "random_strategy_spawn_ticks", "strategy_end_tick"
+    )
+    def test_spawn_at_right_ticks(
+        self,
+        random_strategy: RandomScheduleStrategy,
+        random_strategy_spawn_ticks: list[int],
+        strategy_end_tick: int,
+    ):
+        for tick in range(0, strategy_end_tick * 2):
```

Comment:
why * 2?

Reply:
It's just to ensure that ticks are generated where no trains are allowed to be spawned. Then you can test if there are really no trains spawned.